### DISCUSSION TITLE
Array input

### DISCUSSION THREAD

**Discussion by drmax24** [thecodingmachine/graphqlite#551] [Q&A]
Do we have a simple way to define a simple array of scalars in the input handler?

Here is an example of laravel rebing/graphql query arguments.
I have an argument named **ids** that is an array of ids
I can easily define an array input without creating a custom input type.
```
    public function args(): array
    {
        return [
            'ids'                   => [
                'type' => Type::listOf(Type::int()),
            ],
```

Do we have anything similar?

**Comment by drmax24** [comment]
It kinda works. But how to make ids argument to be optional?
```
    /**
     * @param string[] $ids
     */
    #[Query(
        name: 'groups',
        outputType: '[CommunityGroup]',
    )]
    public function resolve(?array $ids = null, int $offset = 0, int $limit = 10): \Generator
```

ids is not optional in graphql schema

**Comment by oojacoboo** [comment]
@drmax24 in your phpdoc annotation, try defining it like the following...

```php
/**
 * @param string[]|null $ids
 */
```

**Comment by drmax24** [comment]
do i have to use phpdoc for it? is there a way to use annotations?
phpdoc looks outdated

**Reply by oojacoboo** [reply]
This isn’t part of the language, it’s generic typing sugar.  Maybe we get a native generics syntax in PHP 9.  There is an RFC for it that looks promising.
